### DISCUSSION TITLE
Scaling data?

### DISCUSSION THREAD

**Discussion by mmcc-xx** [eclipse-ditto/ditto#1597] [General]
Is there any information out there that talks about Ditto scaling? How storage and RAM requirements change based on the number of twins, the number of properties per twin, and presumably how often those properties change?

**Comment by kalinkostashki** [comment]
Hi there @mmcc-xx ,

actually our team is working on that and will look to commit this to Ditto.
This would include the proper setup to test, our results and configuration based on AWS EKS env.
Hope that this should cover your requirements.

**Reply by mmcc-xx** [reply]
Excellent - I'm very glad to hear that. Do you have a time frame in mind for when this might happen?

**Comment by thjaeckle** [comment]
I would want to state that the resource consumption and scale of the services to apply heavily depends on the things' sizes, message payloads, expected messages to process per second, etc.
And in addition to that to a lot of parameters which can be tuned in Ditto's and Akka's configuration, e.g. Cache sizes, max persistence actors to keep in memory, etc.

So it is always advisable to perform load test for your expected use case and figures you need to achieve.
And to adjust the configuration and resources while optimizing for your scenario.

**Reply by mmcc-xx** [reply]
That makes a lot of sense but at this point I'm just trying to get a rough idea of what the resource needs will look like.
